packages/SwingSet/src/kernel/vat-warehouse.js

```diff
@@ -106,6 +236,41 @@ export function makeVatWarehouse({
     return translators;
   }
 
+  /**
+   * @param {string} vatID
+   * @param {import('../types-external.js').VatKeeper} vatKeeper
+   * @param {VatManager} manager
+   * @returns {Promise<void>}
+   */
+  async function replayTranscript(vatID, vatKeeper, manager) {
+    const snapshotInfo = vatKeeper.getSnapshotInfo();
+    const startPos = snapshotInfo ? snapshotInfo.endPos : undefined;
```

Comment:
yeah this will go away in an upcoming PR, it becomes:
```js
  async function replayTranscript(vatID, vatKeeper, manager) {
    const total = vatKeeper.transcriptSize();
    kernelSlog.write({ type: 'start-replay', vatID, deliveries: total });
    for (const [deliveryNum, te] of vatKeeper.getTranscript()) {
```

The plan is that `replayTranscript` extracts the first entry, which is always either `initialize-worker` or `load-snapshot`. Then it uses that information to call either `vatLoader.initializeWorker` or `vatLoader.loadSnapshot`. And then it replays the remainder of the transcript to the new/reloaded worker. That will require a different relationship between `ensureVatOnline` and `replayTranscript`, but I think it should be cleaner, and the transcript will be a much more literal sequence of instructions for both the normal kernel, and the replay-transcript.js tool.